### PR TITLE
[opt](index compaction)Use RAM dir to create tmp index_writer

### PR DESCRIPTION
**Pull request by qidaye**

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_compaction.cpp [hunk 1]
```diff
@@ -40,8 +40,9 @@ Status compact_column(int64_t index_id, std::vector<lucene::store::Directory*>&
                     "debug point: index compaction error");
         }
     })
-    lucene::store::Directory* dir =
-            DorisFSDirectoryFactory::getDirectory(io::global_local_filesystem(), tmp_path.data());
+    bool can_use_ram_dir = true;
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_compaction.cpp:43]: use be config

Comment (qidaye) [reply be/src/olap/rowset/segment_v2/inverted_index_compaction.cpp:43]: This is the input arg to the getDirectory method.
If only `can_use_ram_dir` and `config::inverted_index_ram_dir_enable` both are true, we can actually use the RAM direcotry.

**Comment by qidaye** [comment]
run buildall

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Pull request merged**